modules/vstudio/vs2010_vcxproj.lua

```diff
@@ -3666,6 +4016,466 @@
+		if _ACTION >= "vs2017" then
+			m.element("ApplicationTypeRevision", nil, "3.0")
+		elseif _ACTION >= "vs2015" then
+			m.element("ApplicationTypeRevision", nil, "2.0")
+		else
+			m.element("ApplicationTypeRevision", nil, "1.0")
+		end
+	end
+
+	function m.androidProjectVersion(cfg)
+		_p(2, "<RootNamespace>%s</RootNamespace>", cfg.project.name)
+		_p(2, "<MinimumVisualStudioVersion>14.0</MinimumVisualStudioVersion>")
+		_p(2, "<ProjectVersion>1.0</ProjectVersion>")
+	end
+
+	function m.androidDebugInformation(cfg)
+		if cfg.flags.Symbols then
+			m.element("GenerateDebugInformation", "true")
+		end
+	end
+
+	function m.androidDebugInformationFormat(cfg)
+		if cfg.symbols ~= "Default" then
+
+			if cfg.symbols == p.OFF then
+				m.element("DebugInformationFormat", nil, "None")
+			elseif cfg.symbols == "Full" then
+				m.element("DebugInformationFormat", nil, "FullDebug")
+			else
+				m.element("DebugInformationFormat", nil, "LineNumber")
+			end
+
+		end
+	end
+
+	function m.androidExceptionHandling(cfg, condition)
+		-- Note: Android defaults to 'off'
+		local exceptions = {
+			On = "Enabled",
+			Off = "Disabled",
+			UnwindTables = "UnwindTables",
+		}
+		if _ACTION >= "vs2015" then
+			if exceptions[cfg.exceptionhandling] ~= nil then
+				m.element("ExceptionHandling", condition, exceptions[cfg.exceptionhandling])
+			end
+		else
+			if cfg.exceptionhandling == premake.ON then
+				m.element("GccExceptionHandling", condition, "true")
+			end
+		end
+	end
+
+	function m.androidRuntimeTypeInfo(cfg, condition)
+		-- Note: Android defaults to 'off'
+		if cfg.rtti == premake.ON then
+			m.element("RuntimeTypeInfo", condition, "true")
+		end
+	end
+
+	function m.androidWarningLevel(cfg, condition)
+
+		if _ACTION >= "vs2015" then
+			if cfg.warnings and cfg.warnings ~= "Off" then
+				m.element("WarningLevel", nil, "EnableAllWarnings")
+			else
+				m.warningLevel(cfg, condition)
+			end
+		else
+			m.warningLevel(cfg, condition)
+		end
+
+	end
+
+	function m.androidClCompilePreprocessorDefinitions(cfg, condition)
+		m.preprocessorDefinitions(cfg, cfg.defines, false, condition)
+	end
+
+	function m.androidLanguageStandardCpp(cfg)
+		local cpp_langmap = {
+			["C++98"]   = "c++98",
+			["C++11"]   = "c++11",
+			["C++14"]   = "c++1y",
+			["C++17"]   = "c++1z",
+			["C++latest"] = "c++1z",
+			["gnu++98"] = "gnu++98",
+			["gnu++11"] = "gnu++11",
+			["gnu++14"] = "gnu++1y",
+			["gnu++17"] = "gnu++1z",
+		}
+		if cpp_langmap[cfg.cppdialect] ~= nil then
+			m.element("CppLanguageStandard", nil, cpp_langmap[cfg.cppdialect])
+		end
+	end
+
+	function m.androidLanguageStandardC(cfg)
+		local c_langmap = {
+			["C98"]   = "c98",
+			["C99"]   = "c99",
+			["C11"]   = "c11",
+			["gnu99"] = "gnu99",
+			["gnu11"] = "gnu11",
+		}
+		if c_langmap[cfg.cdialect] ~= nil then
+			m.element("CLanguageStandard", nil, c_langmap[cfg.cdialect])
+		end
+	end
+
+	function m.androidAdditionalCompileOptions(cfg)
+
+		if _ACTION >= "vs2015" then
+
+		else
```

Comment:
Missing code?
else `if _ACTION < "vs2015" then`

Reply:
Part of the original code